docker/dev_env/run.sh

```diff
@@ -0,0 +1,79 @@
+#! /usr/bin/env bash
+
+set -e
+
+# Kudos https://stackoverflow.com/a/10910180 for macOS group ID reading
+case "$OSTYPE" in
+linux*)
+  host_docker_gid="$(getent group docker | cut -d: -f3)"
+  ;;
+
+darwin*)
+  host_docker_gid="$(dscl . -read /Groups/docker | awk '($1 == "PrimaryGroupID:") { print $2 }')"
```

Comment:
There is no `docker` group in macOS. Most of the permission-related stuff is handled by Docker Desktop or Orbstack. #4401 takes a userless approach that works well on a Mac but can, understandably, be against what I assume were your goals for this PR.